monetio/sat/_omps_nadir_mm.py

```diff
@@ -13,30 +13,36 @@ def read_OMPS_nm(files):
     import xarray as xr
 
     count = 0
-    print(files)
+    #print(files)
     # Check if files are url
     if "https" in files[0]:
         filelist = sorted(files)
         for filename in filelist:
             data = extract_OMPS_nm_opendap(filename)
-            # print(data)
             if count == 0:
                 data_array = data
                 count += 1
             else:
                 data_array = xr.concat([data_array, data], "x")
     else:
-        filelist = sorted(glob(files, recursive=False))
-
+        if isinstance(files,str):
+            filelist = sorted(glob(files, recursive=False))
+        else: 
+            filelist = sorted(files) # assume list
         for filename in filelist:
-            data = extract_OMPS_nm(filename)
-
-            if count == 0:
-                data_array = data
-                count += 1
-            else:
-                data_array = xr.concat([data_array, data], "x")
-
+            try:
+                data = extract_OMPS_nm(filename)
+                if count == 0:
+                    data_array = data
+                    count += 1
+                else:
+                    data_array = xr.concat([data_array, data], "x")
```

Comment:
`count` functions as a flag for initializing the combined dataset for all observations to be read in. Unless I'm mistaken, in its current location it only increments for the first file successfully read in.

Reply:
Ah yes I see what you're saying. A better option might be to have a list of data arrays that you append to upon successful load, then just call xarray concat once at the end. Then `count` isn't needed.

Reply:
But fine to leave it this way too.

Reply:
@zmoon: I think that this may need to stay how it is until after the tutorial. The ValueError that comes up for files with weird cross-track dimensions occurs in the xarray concat, so appending on successful load and then calling xarray concat at the very end results in passing no data out of the reader if there is a ValueError.

Reply:
Would you mind adding some comments about the `KeyError` and `ValueError` situations under the `except` line?

Reply:
Added.

https://github.com/mbruckner-work/monetio/blob/8c027520e7ec61f111aa524594abc129a0cd4401/monetio/sat/_omps_nadir_mm.py#L39-L42

Reply:
Great, thanks